tree/30_generic_methods/variable_dict.cf

```diff
@@ -0,0 +1,52 @@
+#####################################################################################
+# Copyright 2013 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name Variable dict
+# @description Define a variable that contains key,value pairs (a dictionnary)
+#
+# @parameter variable_prefix The prefix of the variable name, use ${class_prefix} for a local variable and a static name for a global variable
```

Comment:
It is a variable defined by the technique editor that is defined in the same way as in methods ie:
canonify(join("_", "this.callers_promisers"))